From b30e244a02d159be7789a08b6c65793f472415f7 Mon Sep 17 00:00:00 2001
From: Evelyn Tsai <evelyn.tsai@mediatek.com>
Date: Mon, 14 Sep 2026 14:43:11 +0800
Subject: [PATCH] mtk: wifi: mt76: mt7915: Fixed null pointer dereference issue

Without this patch, when the station is still in Authentication stage
and sends a "Notify bandwidth change action frame" to AP at the same
time, there will be a race condition that causes a crash to occur
because the AP access "msta->vif" that has not been fully initialized.

Signed-off-by: Bo Jiao <Bo.Jiao@mediatek.com>
Signed-off-by: Money Wang <money.wang@mediatek.com>
Signed-off-by: MeiChia Chiu <meichia.chiu@mediatek.com>
Signed-off-by: Evelyn Tsai <evelyn.tsai@mediatek.com>
---
 mt7915/main.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/mt7915/main.c b/mt7915/main.c
index ae1656107..066f8d85c 100644
--- a/mt7915/main.c
+++ b/mt7915/main.c
@@ -1411,6 +1411,12 @@ static void mt7915_sta_rc_update(struct ieee80211_hw *hw,
 	if (!msta->wcid.sta)
 		return;
 
+	if (!msta->vif) {
+		dev_warn(dev->mt76.dev, "Un-initialized STA %pM wcid %d in rc_work\n",
+			 sta->addr, msta->wcid.idx);
+		return;
+	}
+
 	mt7915_sta_rc_work(&changed, sta);
 	ieee80211_queue_work(hw, &dev->rc_work);
 }
-- 
2.45.2

